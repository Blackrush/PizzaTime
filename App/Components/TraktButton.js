import React, { Component } from 'react'
import {
  StyleSheet,
  TouchableHighlight,
  Text,
} from 'react-native'


class TraktButton extends Component {
  static propTypes = {
    label: React.PropTypes.string.isRequired,
    onPress: React.PropTypes.func,
  }

  render() {
    return (
      <TouchableHighlight underlayColor="#c61017" style={styles.button} onPress={this.props.onPress}>
        <Text style={styles.label}>{this.props.label.toUpperCase()}</Text>
      </TouchableHighlight>
    )
  }
}

const styles = StyleSheet.create({
  button: {
    backgroundColor: '#ed1c24',
    borderColor: '#de1219',
    borderWidth: 1,
    paddingTop: 10,
    paddingBottom: 10,
    paddingLeft: 16,
    paddingRight: 16,
  },
  label: {
    color: '#fff',
    fontFamily: 'PingFangTC-Regular',
  },
})

export default TraktButton