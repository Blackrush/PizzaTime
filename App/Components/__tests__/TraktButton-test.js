import React from 'react'
import TraktButton from '../TraktButton'
import { create as render } from 'react-test-renderer'

test('TraktButton renders correctly', () => {
  const tree = render(
    <TraktButton label="Login with Trakt.tv"/>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})
