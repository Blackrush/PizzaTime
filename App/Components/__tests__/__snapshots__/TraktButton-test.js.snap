exports[`test TraktButton renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function bound touchableHandleResponderGrant]}
  onResponderMove={[Function bound touchableHandleResponderMove]}
  onResponderRelease={[Function bound touchableHandleResponderRelease]}
  onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
  onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
  onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent"
      },
      Object {
        "backgroundColor": "#ed1c24",
        "borderColor": "#de1219",
        "borderWidth": 1,
        "paddingBottom": 10,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 10
      }
    ]
  }
  testID={undefined}>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontFamily": "PingFangTC-Regular"
      }
    }>
    LOGIN WITH TRAKT.TV
  </Text>
</View>
`;
